api/src/main/java/org/pmiops/workbench/profile/ProfileService.java

```diff
@@ -269,6 +262,34 @@ public DbUser updateProfile(
     return updatedUser;
   }
 
+  private void populateDemographicSurvey(
+      DbUser user, Profile updatedProfile, Timestamp completionTime) {
+    DbDemographicSurvey dbDemographicSurvey =
+        demographicSurveyMapper.demographicSurveyToDbDemographicSurvey(
+            updatedProfile.getDemographicSurvey());
+    if (user.getDemographicSurveyCompletionTime() == null && dbDemographicSurvey != null) {
+      user.setDemographicSurveyCompletionTime(completionTime);
+    }
+    if (dbDemographicSurvey != null && dbDemographicSurvey.getUser() == null) {
+      dbDemographicSurvey.setUser(user);
+    }
+    user.setDemographicSurvey(dbDemographicSurvey);
+  }
+
+  private void populateDemographicSurveyV2(DbUser dbUser, Profile updatedProfile) {
+    DbDemographicSurveyV2 newDemoSurvey =
+        demographicSurveyMapper.toDbDemographicSurveyV2(updatedProfile.getDemographicSurveyV2());
+
+    if (newDemoSurvey != null) {
+      DbDemographicSurveyV2 existingSurvey = dbUser.getDemographicSurveyV2();
+      if (existingSurvey != null) {
+        newDemoSurvey.setId(existingSurvey.getId());
```

Comment:
here is the fix to the problem I was seeing - I needed to set the ID of the `newDemoSurvey` to be equal to existing row ID, so it knows to update it in-place instead of attempting to create a new one